Math/NumberTheory/ArithmeticFunctions/Standard.hs

```diff
@@ -149,6 +151,30 @@ jordanHelper pa 2 = (pa - 1) * pa
 jordanHelper pa k = (pa - 1) * pa ^ wordToInt (k - 1)
 {-# INLINE jordanHelper #-}
 
+ramanujan :: (UniqueFactorisation n, Integral n, Integral m) => n -> m
+ramanujan = runFunction ramanujanA
+
+-- | Calculates the <https://en.wikipedia.org/wiki/Ramanujan_tau_function Ramanujan tau function>
+--   of a positive number @n@, using formulas given <http://www.numbertheory.org/php/tau.html here>
+ramanujanA :: forall n m. (UniqueFactorisation n, Integral n, Integral m) => ArithmeticFunction n m
+ramanujanA = multiplicative $ \((unPrime :: Prime n -> n) -> p) -> ramanujanHelper p
+
+ramanujanHelper :: (UniqueFactorisation n, Integral n, Integral m) => n -> Word -> m
+ramanujanHelper _ 0 = 1
+ramanujanHelper 2 1 = -24
+ramanujanHelper p 1 = (65 * sigmaHelper (p ^ (11 :: Int)) 1 + 691 * sigmaHelper (p ^ (5 :: Int)) 1 - 691 * 252 * 2 * sum [sigma 5 k * sigma 5 (p-k) | k <- [1..(p `quot` 2)]]) `quot` 756
+ramanujanHelper p k = sum $ zipWith3 (\a b c -> a * b * c) paPowers tpPowers binomials
+  where pa = fromIntegral $ p ^ (11 :: Int)
+        tp = ramanujanHelper p 1
+        paPowers = iterate (* (-pa)) 1
+        binFunc j
+          | 3*j > k   = row !! (wordToInt $ k - 2*j)
+          | otherwise = row !! (wordToInt $ j)
+          where row = binomial !! (wordToInt $ k - j)
+        binomials = map binFunc [0 .. k `quot` 2]
```

Comment:
I came up with an idea to avoid random access to the list of binomials. On my machine it is 5x faster in `ArithmeticFunctions/ramanujan prime power` benchmark.

```haskell
        binomials = scanl
          (\acc j -> acc * (k' - 2 * j) * (k' - 2 * j - 1) `quot` (k' - j) `quot` (j + 1))
          1
          [0 .. k' `quot` 2 - 1]
        k' = fromIntegral k 
```

Reply:
Lovely! Nicely done.